src/sounds.cpp

```diff
@@ -204,12 +209,21 @@ void sounds::process_sound_markers( player *p )
             continue;
         }
 
+        // At this point we are dealing with attention (as opposed to physical effects)
+        // so reduce volume by the amount of ambient noise from the weather.
+        const int mod_vol = volume - weather_vol;
+
+        // The noise was drowned out by the surroundings
+        if (mod_vol <= 0) {
```

Comment:
Ah I had missed that. I'll probably do the latter, since you shouldn't be hearing sounds that were dropped to 0.